docs/graph-algorithms/breadth-first-search.mdx

```diff
@@ -0,0 +1,416 @@
+---
+title: Breadth First Search
+slug: /graph-algorithms/breadth-first-search
+---
+
+## Introduction
+Breadth-first search (BFS) is an algorithm for traversing or searching tree or graph data structures.
+It starts at the tree root (or some arbitrary node of a graph, sometimes referred to as a 'search key'[1]),
+and explores all of the neighbor nodes at the present depth prior to moving on to the nodes at the next depth level.
+
+## Explanation
+
+- Given an undirected & Connected Graph (V, E) <br />
+  V: Vertices E: Edges <br />
+- Goal: methodically explore every vertex and every edge <br />
+- “Explore” a graph, turning it into a tree<br />
+- One vertex at a time<br />
+- Expand frontier of explored vertices across its breadth<br />
+- Discovers all vertices at distance k from s before those at distance k+1
+- A level-by-level order traversal<br />
+- For example, in the following graph, we start traversal from vertex 2. When we come to vertex 0, we look for all adjacent vertices of it. 2 is also an adjacent vertex of 0. If we don’t mark visited vertices, then 2 will be processed again and it will become a non-terminating process.
+ A Breadth First Traversal of the following graph is 2, 0, 3, 1.<br />
+ [![graph.png](https://media.geeksforgeeks.org/wp-content/uploads/bfs-5.png)]<br />
```

Comment:
Do not use any image by geeks for geeks, we only consider original images in contribution. You can use [this](https://graphonline.ru/en/) to create your own graph.